### PR TITLE
feat: add possibility to enable/disable exclusive queue feature

### PR DESCRIPTION
**Pull request by lucasssouza1**
## What is being delivered?

- Adds a new setting option (STOMP_DEFAULT_EXCLUSIVE_QUEUE) that allows the app to create a queue in RabbitMQ as exclusive or not.

## What impacts?

When the setting `STOMP_DEFAULT_EXCLUSIVE_QUEUE` is disable (`False`), the queues created by consumers won't be `exclusive`. 

## Examples

STOMP_DEFAULT_EXCLUSIVE_QUEUE == `False`
1-exclusive-false

STOMP_DEFAULT_EXCLUSIVE_QUEUE == `True`
2-exclusive-true

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -156,6 +156,10 @@ defined, this parameter **must** be an integer!
 
   Optional parameter that controls how many seconds django-stomp will wait for a message to be processed. Defaults to 30 seconds. If defined, this parameter **must** be an integer!
 
+***STOMP_DEFAULT_EXCLUSIVE_QUEUE***
+
+  Optional parameter that defines if the default value of `exclusive` queue parameter will be True or False in RabbitMQ. The default is `False`.
```
Comment (niltonfrederico) [inline README.md:161]: [suggestion]
We can be more explicit when this is applied:

```suggestion
  Optional parameter that defines the default value of `exclusive` queue parameter will be True or False when creating a queue in RabbitMQ. The default is `False`.
```

Comment (lucasssouza1) [reply README.md:161]: Good point! I'll fix it!

File: tests/unit/services/test_consumer.py [hunk 2]
```diff
@@ -79,3 +79,29 @@ def test_should_have_only_one_django_stomp_listener_even_if_set_listener_is_call
...
+
+    # Assert - the created headers setup must have the same mapped headers keys
+    assert headers_setup_mapping.keys() == created_header_setup.keys()
+
+    # Assert - the created headers setup items must have the types porperly configured.
+    for key, value in headers_setup_mapping.items():
+        assert isinstance(created_header_setup[key], value)
+
+    # Assert - it should be possible to parse the value 'prefetch-count' to int
+    assert int(created_header_setup["prefetch-count"])
```
Comment (willianantunes) [inline tests/unit/services/test_consumer.py:107]: **suggestion**: We can remove the comments and apply them at the assert statements.

```suggestion
    # Arrange - build listener to some arbirtrary queue and get its connection configuration headers
    django_stomp_listener = builder.build_listener(f"some-destination-{uuid4()}")
    created_header_setup = django_stomp_listener._connection_configuration["headers"]

    assert headers_setup_mapping.keys() == created_header_setup.keys(), "headers do not have all required keys"
    for key, value in headers_setup_mapping.items():
        assert isinstance(created_header_setup[key], value), "header has wrong type"
    assert int(created_header_setup["prefetch-count"]), "prefetch-count must be a number"
```

**Review by willianantunes — APPROVED** [review]

File: tests/unit/services/test_consumer.py (hunk 2 above)
Comment (lucasssouza1) [reply tests/unit/services/test_consumer.py:107]: It's a good ideia. I'll accept your suggestion.

Comment (lucasssouza1) [reply tests/unit/services/test_consumer.py]: Thanks for the review!

**Review by abxsantos — APPROVED** [review]

File: django_stomp/services/consumer.py [hunk 3]
```diff
@@ -207,6 +208,8 @@ def build_listener(
         # These two parameters must be set on producer side as well, otherwise you'll get precondition_failed
         "x-dead-letter-routing-key": create_dlq_destination_from_another_destination(destination_name),
         "x-dead-letter-exchange": "",
+        # This parameter below defines if the queue to be created will be exclusive or not.
+        "exclusive": STOMP_DEFAULT_EXCLUSIVE_QUEUE,
```
[the remark this answers is not in the document]
Comment (lucasssouza1) [reply django_stomp/services/consumer.py:212]: Good suggestion, @abxsantos !

> What do you think about adding tests to ensure the correct behavior is achieved?

Added one more unit test to verify parameter change behavior

> Also, from what I remember, the publish step will also create a new queue, if it doesn't already exists. Did you check this as well?

In this case I just test the module behavior using a local django-stomp with a local django app. The evidences are on the PR.

**Comment by lucasssouza1** [comment]
This PR was closed because the feature was added [here](https://github.com/juntossomosmais/django-stomp/pull/61)

**Pull request closed without merging**